### PR TITLE
Add VersionService

### LINKED ISSUE #279: Add VersionService
Convenient service to find out the app version that is set in the manifest/plist

### PR DESCRIPTION
**Pull request by jperedadnr**
Fixes #279

### CODE REVIEW

File: modules/version/src/main/java/com/gluonhq/attach/version/impl/DummyVersionService.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.gluonhq.attach.version.impl;
+
+import com.gluonhq.attach.version.VersionService;
+
+public abstract class DummyVersionService implements VersionService {
```
Comment (johanvos) [inline modules/version/src/main/java/com/gluonhq/attach/version/impl/DummyVersionService.java:32]: This is not used anywhere?

File: modules/version/src/main/native/android/c/version.c [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+JNIEXPORT jstring JNICALL Java_com_gluonhq_attach_version_impl_AndroidVersionService_nativeVersion
+(JNIEnv *env, jclass jClass)
+{
+    if (isDebugAttach()) {
+        ATTACH_LOG_FINE("Retrieving Version number\n");
+    }
+    ATTACH_DALVIK();
+    jstring version = (*dalvikEnv)->CallObjectMethod(dalvikEnv, jDalvikVersionService, jVersionServiceGetVersion);
+    const char *responseVersionChars = (*dalvikEnv)->GetStringUTFChars(dalvikEnv, version, 0);
```
Comment (johanvos) [inline modules/version/src/main/native/android/c/version.c:81]: here and on other places we should later tell dalvikEnv that we don't need access to responseVersionChars anymore. It's tricky, as we might still need it in the regular env. This is probably why we don't have it in most of the services, but it introduces a memory leak.
Would be good to file a separate issue for this, but I'm ok with keeping it like this for now.

File: modules/version/src/main/java/com/gluonhq/attach/version/impl/DummyVersionService.java (hunk 1 above)
Comment (jperedadnr) [reply modules/version/src/main/java/com/gluonhq/attach/version/impl/DummyVersionService.java:32]: It is needed with the non-platform jar. 
For every service, we do `exports com.gluonhq.attach.XXX.impl to com.gluonhq.attach.util;` in the module-info class, and for that we need a non empty package.

File: modules/version/src/main/native/android/c/version.c (hunk 2 above)
Comment (jperedadnr) [reply modules/version/src/main/native/android/c/version.c:81]: Filed: https://github.com/gluonhq/attach/issues/281

**Review by johanvos — APPROVED** [review]

**Pull request merged**